backend/fms_core/template_importer/row_handlers/project_link_samples/project_link_samples.py

```diff
@@ -0,0 +1,55 @@
+from datetime import datetime
+
+from fms_core.models import SampleByProject
+
+from fms_core.template_importer.row_handlers._generic import GenericRowHandler
+
+from fms_core.services.sample import get_sample_from_container
+from fms_core.services.project import get_project
+from fms_core.services.project_link_samples import create_link, remove_link
+
+ADD_ACTION = "ADD"
+REMOVE_ACTION = "REMOVE"
+
+class ProjectLinkSamplesHandler(GenericRowHandler):
+    def __init__(self):
+        super().__init__()
+
+
+    def process_row_inner(self, sample, project, action):
+        print('start project link samples row handler')
+
+        # Get sample object
+        sample_obj, self.errors['sample'], self.warnings['sample'] = get_sample_from_container(barcode=sample["sample_container_barcode"],
+                                                                                               coordinates=sample["sample_container_coord"])
+        # Get project object
+        project_obj, self.errors['project'], self.warnings['project'] = get_project(project['name'])
+
+        if not sample_obj or not project_obj:
+            self.errors['link'] = (f"Unable to process sample or project information.")
+            return
+
+        link_exists = SampleByProject.objects.filter(sample=sample_obj, project=project_obj).exists()
+
+        # Check if link exists to ensure there's not a duplicated association
+        if action['name'] == ADD_ACTION:
+            if link_exists:
+                self.errors['link'] = (f"[Sample {sample['sample_name']}] is already associated to project [{project['name']}].")
+            else:
+                # Create link object if no errors
+                project_sample_link, self.errors['link'], self.warnings['link'] = create_link(sample=sample_obj,
+                                                                                              project=project_obj)
+
+        # If the link doesn't exists we can't perform a remove action
+        elif action['name'] == REMOVE_ACTION:
+            if not link_exists:
+                self.errors['link'] = (f"Sample [{sample['sample_name']}] is not currently associated to project [{project['name']}].")
+            else:
+                # Remove link object if no errors
+                num_deleted, self.errors['link'], self.warnings['link'] = remove_link(sample=sample_obj,
+                                                                                      project=project_obj)
```

Comment:
The error handling logic should be done at the level of the Services. Those would return the appropriate warnings and/or errors to the RowHandler